### PR TITLE
test: add coverage tests for rabitq_split, mmap_io, warp, and c_api

### PR DESCRIPTION
**Pull request by LHT129**
## Summary
- Add unit tests for RaBitQSplitDataCell: InsertVector, UpdateVector, Serialize/Deserialize, Encode/Decode, Resize, ShrinkToFit, Move, ExportModel, MergeOther, QueryWithDistanceFilter, IP metric
- Add unit tests for MMapIO: directory path error handling, resize shrink, MultiRead, existing file reopen
- Add functional test for Warp: multiple dimensions with IP metric
- Existing vsag_c_api tests: null handles, invalid inputs, update failures, serialize edge cases

Addresses the main branch coverage gap (currently 87.92%, target 90%). Focuses on the highest-miss files identified via Codecov: rabitq_split_datacell.h (231 miss), warp.cpp (107 miss), mmap_io.cpp (43 miss), vsag_c_api.cpp (91 miss).

## Test plan
- [x] All new tests pass locally (debug build)
- [ ] CI coverage workflow runs and shows coverage improvement
- [ ] No regressions in existing tests

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/vsag_c_api_test.cpp [hunk 1]
```diff
@@ -417,3 +420,218 @@ TEST_CASE("vsag_c_api update operations and get vector by ids", "[vsag_c_api][ut
...
+    REQUIRE(vsag_index_knn_search_with_filter(
+                nullptr, test_case.datas.data(), dim, 3, hgraph_search_parameters, nullptr, &result)
+                .code == VSAG_SUCCESS);
+    REQUIRE(vsag_index_range_search(
+                nullptr, test_case.datas.data(), dim, 1.0F, hgraph_search_parameters, &result)
+                .code == VSAG_SUCCESS);
+    REQUIRE(
+        vsag_index_range_search_with_filter(
+            nullptr, test_case.datas.data(), dim, 1.0F, hgraph_search_parameters, nullptr, &result)
+            .code == VSAG_SUCCESS);
```
[the remark this answers is not in the document]
Comment (LHT129) [reply src/vsag_c_api_test.cpp]: Fixed in 7a520133. Removed the assertions that expected VSAG_SUCCESS on null handles — these API functions silently skip when index is null (no segfault), but asserting success was misleading. Now we just call them without asserting success to exercise the null-guard code paths.

File: src/datacell/flatten_datacell_test.cpp [hunk 2]
```diff
@@ -145,3 +145,216 @@ TEST_CASE("RaBitQSplitDataCell direct split compute", "[ut][RaBitQSplitDataCell]
...
+
+    SECTION("ComputePairVectors") {
+        flatten->BatchInsertVector(vectors.data(), count);
+        bool need_release = false;
+        const auto* code0 = flatten->GetCodesById(0, need_release);
+        REQUIRE(code0 != nullptr);
+        if (need_release) {
+            flatten->Release(code0);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (LHT129) [reply src/datacell/flatten_datacell_test.cpp]: Fixed in 7a520133. Renamed section to "GetCodesById" since that is what it actually tests. ComputePairVectors cannot be called on RaBitQSplitDataCell (the quantizer throws "building the index is not supported using RabbitQ alone" when calling Compute on code pairs).

File: src/datacell/flatten_datacell_test.cpp [hunk 3]
```diff
@@ -145,3 +145,216 @@ TEST_CASE("RaBitQSplitDataCell direct split compute", "[ut][RaBitQSplitDataCell]
...
+
+    SECTION("Encode and Decode") {
+        flatten->BatchInsertVector(vectors.data(), count);
+        auto code_size = flatten->code_size_;
+        std::vector<uint8_t> codes(code_size);
+        REQUIRE(flatten->Encode(vectors.data(), codes.data()) == true);
+        std::vector<float> decoded(dim);
+        flatten->Decode(codes.data(), decoded.data());
+    }
```
[the remark this answers is not in the document]
Comment (LHT129) [reply src/datacell/flatten_datacell_test.cpp:241]: Fixed in 7a520133. RaBitQ Decode returns false (lossy quantization does not support full decode), so we call it without asserting true. The Encode assertion remains.

File: src/io/mmap_io_test.cpp [hunk 4]
```diff
@@ -60,3 +60,75 @@ TEST_CASE("MMapIO Serialize & Deserialize", "[ut][MMapIO]") {
...
+    REQUIRE(io->size_ >= 8192);
+
+    io->Resize(2048);
+    REQUIRE(io->size_ == 2048);
+
+    std::vector<uint8_t> read_buf(2048);
+    io->Read(2048, 0, read_buf.data());
+    for (uint64_t i = 0; i < 2048; ++i) {
+        REQUIRE(read_buf[i] == 0xAB);
+    }
```
[the remark this answers is not in the document]
Comment (LHT129) [reply src/io/mmap_io_test.cpp:90]: Fixed in 7a520133. Added REQUIRE on the Read return value.

File: src/vsag_c_api_test.cpp [hunk 5]
```diff
@@ -417,3 +420,218 @@ TEST_CASE("vsag_c_api update operations and get vector by ids", "[vsag_c_api][ut
...
+    REQUIRE(ret.code != VSAG_SUCCESS);
+
+    deserialize_read_func_path = path;
+
+    using SizeFuncType = SizeType (*)();
+    SizeFuncType size_func = []() -> SizeType {
+        struct stat st;
+        stat(deserialize_read_func_path.data(), &st);
+        return static_cast<SizeType>(st.st_size);
+    };
```
[the remark this answers is not in the document]
Comment (LHT129) [reply src/vsag_c_api_test.cpp:612]: Fixed in 7a520133. Initialized stat struct with {} and added REQUIRE on the stat() return value.

**Review by wxyucs — APPROVED** [review]
lgtm
